string Class
float64 probability
int64 x
int64 y
int64 z

